packages/web3-eth/test/unit/utils/watch_transaction_by_subscription.test.ts

```diff
@@ -14,117 +14,124 @@ GNU Lesser General Public License for more details.
+	const CONFIRMATION_BLOCKS = 5;
 	describe('should revert to polling in cases where getting by subscription did not workout', () => {
 		let web3Context: Web3Context<Web3EthExecutionAPI>;
 
 		beforeEach(() => {
-			jest.spyOn(Web3RequestManager.prototype, 'send').mockImplementation(async () => {
-				return {} as Promise<unknown>;
-			});
-			jest.spyOn(WebSocketProvider.prototype, 'request').mockImplementation(async () => {
-				return {} as Promise<JsonRpcResponseWithResult<unknown>>;
-			});
-
-			(ethRpcMethods.sendRawTransaction as jest.Mock).mockResolvedValue(
-				expectedTransactionHash,
-			);
-			(ethRpcMethods.getTransactionReceipt as jest.Mock).mockResolvedValue(
-				expectedTransactionHash,
-			);
 			web3Context = new Web3Context({
-				// dummy provider that does supports subscription
-				provider: new WebSocketProvider('ws://localhost:8546'),
-				registeredSubscriptions,
-			});
+			 	provider: new WebSocketProvider('wss://localhost:8546'),}
+		);
+
 			(web3Context.provider as any).supportsSubscriptions = () => true;
+			web3Context.transactionConfirmationBlocks = CONFIRMATION_BLOCKS;
+			web3Context.enableExperimentalFeatures.useSubscriptionWhenCheckingBlockTimeout =
+			true;
+
 		});
-		afterEach(() => {
-			// to clear the interval inside the subscription function:
-			web3Context.transactionConfirmationBlocks = 0;
-		});
-		let counter = 0;
-		it.each(testData)(
-			`should call getBlockNumber if blockHeaderTimeout reached\n ${testMessage}`,
-			async (_, inputTransaction) => {
-				if (counter > 0) {
-					return;
-				}
-				counter += 1;
-				const formattedTransactionReceipt = format(
-					transactionReceiptSchema,
-					expectedTransactionReceipt,
-					DEFAULT_RETURN_FORMAT,
-				);
 
-				web3Context.enableExperimentalFeatures.useSubscriptionWhenCheckingBlockTimeout =
-					true;
-				// this will case the function to revert to polling:
-				web3Context.blockHeaderTimeout = 0;
+		it.each(testData)(
+			`should call getBlockByNumber if blockHeaderTimeout reached\n ${testMessage}`,
+			async (_, inputTransaction,) => {
 
-				web3Context.transactionSendTimeout = 2;
+				let blockNum = 100;
+				let ethGetBlockByNumberCount = 0;
+				web3Context.requestManager.send = jest.fn(async (request) => {
+	
+					if (request.method === 'eth_getBlockByNumber') {
+						ethGetBlockByNumberCount += 1;
+						return Promise.resolve( 
+							{	...blockMockResult.result, 
+								number: (request as any).params[0]
+							});
+					}
+					if (request.method === 'eth_call') {
+	
+						return Promise.resolve("0x");
+					}
+					if (request.method === 'eth_blockNumber') {
+						blockNum += 1;
+						return Promise.resolve(blockNum.toString(16));
+					}
+					if (request.method === 'eth_sendRawTransaction') {
+						return Promise.resolve(expectedTransactionHash);
+					}
+					if (request.method === 'eth_getTransactionReceipt') {
+						return Promise.resolve(expectedTransactionReceipt);
```

Comment:
there is no eth_subscribe added here so subscription fails & lib falls back to polling and will call eth_getBlockByNumber  multiple times